src/backoffice/shared/components/debug/debug.modal.element.ts

```diff
@@ -0,0 +1,163 @@
+import { css, html, nothing, TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { UUITextStyles } from '@umbraco-ui/uui-css';
+import { UmbModalHandler, UmbModalLayoutElement } from '@umbraco-cms/modal';
+
+@customElement('umb-debug-modal-layout')
+export class UmbDebugModalLayout extends UmbModalLayoutElement {
+	static styles = [
+		UUITextStyles,
+		css`
+			uui-dialog-layout {
+				display: flex;
+				flex-direction: column;
+				height: 100%;
+
+				padding: var(--uui-size-space-5);
+				box-sizing: border-box;
+			}
+
+			uui-scroll-container {
+				overflow-y: scroll;
+				max-height: 100%;
+				min-height: 0;
+				flex: 1;
+			}
+
+			uui-icon {
+				vertical-align: text-top;
+				color: var(--uui-color-danger);
+			}
+
+			.context {
+				padding:15px 0;
+				border-bottom:1px solid var(--uui-color-danger-emphasis);
+			}
+
+			h3 {
+				margin-top: 0;
+				margin-bottom: 0;
+			}
+
+			h3 > span {
+				border-radius: var(--uui-size-4);
+				background-color: var(--uui-color-danger);
+				color: var(--uui-color-danger-contrast);
+				padding: 8px;
+				font-size: 12px;
+			}
+			
+
+			ul {
+				margin-top: 0;
+			}
+		`,
+	];
+
+
+	// the modal handler will be injected into the element when the modal is opened.
+	@property({ attribute: false })
+	modalHandler?: UmbModalHandler;
+
+	private _handleClose() {
+		/* Optional data of any type can be applied to the close method to pass it
+		   to the modal parent through the onClose promise. */
+		//this.modalHandler?.close('MY DATA');
+		this.modalHandler?.close();
+	}
+
+	render() {
+		return html`
+			<uui-dialog-layout>
+				<span slot="headline">
+					<uui-icon name="umb:bug"></uui-icon> Debug: Contexts
+				</span>
+				<uui-scroll-container id="field-settings">
+					${this._renderContextAliases()}
+				</uui-scroll-container>
+				<uui-button slot="actions" look="primary" label="Close sidebar" @click="${this._handleClose}">Close</uui-button>
+			</uui-dialog-layout>
+		`;
+	}
+
+	private _renderContextAliases() { 
+		if(!this.data) {
+			return nothing;
+		}
+
+		const aliases = [];
+		for (const [alias, instance] of this.data.contexts) {
+			aliases.push(
+				html`
+				<div class="context">
+					<h3>${alias} <span>${typeof instance}</span></h3>
+					${this._renderInstance(instance)}
```

Comment:
Consider letting this modal layout render `this.data?.content` and reuse the same templates from the outer DebugElement class instead. This way you avoid duplicating the logic. See ModalLayoutConfirmElement for inspiration.

Alternatively, consider putting the rending into a new element (UmbDebugRenderElement) and use that for both inline and modal modes.

```
<slot></slot>
```

Reply:
I have done as you suggested and made it more generic and now pass the `TemplateResult` as a property inside the data object of the dialog/modal.